sdk/python/kfp/components/executor.py

```diff
@@ -182,7 +183,7 @@ def _write_executor_output(self, func_output: Optional[Any] = None):
                   len(self._return_annotation._fields), self._func.__name__,
                   len(func_output)))
         for field in self._return_annotation._fields:
-          field_type = self._return_annotation._field_types[field]
+          field_type = self._return_annotation.__annotations__[field]
```

Comment:
Is this compatible with both Python v3.6 and v3.9?

Reply:
Yes, I believe it is.